datafusion/common/src/config.rs

```diff
@@ -1421,6 +1511,84 @@ impl TableParquetOptions {
     }
 }
 
+#[cfg(feature = "parquet")]
+impl TryFrom<&TableParquetOptions> for WriterPropertiesBuilder {
```

Comment:
This file is already pretty large -- could you move the code into `datafusion/common/src/file_options/parquet_writer.rs` instead? I think since `TableParquetOptions` has all pub fields, this would be feasible

Also, doing that would leave the configuration settings independent of parquet (long term we are hoping to make it easier to have parquet enabled/disabled)